probe-rs/src/debug/unit_info.rs

```diff
@@ -17,30 +24,48 @@ pub(crate) struct UnitInfo<'debuginfo> {
 impl<'debuginfo> UnitInfo<'debuginfo> {
     /// Get the DIE for the function containing the given address.
     ///
+    /// If `stackframe_registers` is not `None`, then the function DIE's will have valid frame_base values calculated from the `DW_AT_frame_base` attribute.
     /// If `find_inlined` is `false`, then the result will contain a single [`FunctionDie`]
     pub(crate) fn get_function_dies(
         &self,
-        address: RegisterValue,
+        address: u64,
+        stackframe_registers: Option<&DebugRegisters>,
         find_inlined: bool,
     ) -> Result<Vec<FunctionDie>, DebugError> {
         log::trace!("Searching Function DIE for address {}", address);
 
         let mut entries_cursor = self.unit.entries();
-
         while let Ok(Some((_depth, current))) = entries_cursor.next_dfs() {
             if current.tag() == gimli::DW_TAG_subprogram {
                 let mut ranges = self.debug_info.dwarf.die_ranges(&self.unit, current)?;
 
                 while let Ok(Some(ranges)) = ranges.next() {
-                    if (RegisterValue::from(ranges.begin) <= address)
-                        && (address < RegisterValue::from(ranges.end))
-                    {
+                    if ranges.begin <= address && address < ranges.end {
```

Comment:
```suggestion
                    if ranges.contains(address) {
```

Reply:
@Yatekii ... I don't understand this one. In this case, `Range` is a `gimli::Read::Range` and does not have a contains method.

Reply:
Ohhh I did not realize! Nevermind then!